src/components/SvgEdit.tsx

```diff
@@ -1,707 +0,0 @@
-    if (shift) {
-      points.push(getMouseCoords(e, true));
-      if (points.length > 1) {
-        const a = points[points.length - 2];
-        const b = points[points.length - 1];
-        const distance = Math.hypot(a.x - b.x, a.y - b.y);
-        if (distance > 130) {
-          // Convert from 'Line to' to 'Curve to'
-          const deltaX = Math.round((b.x - a.x) / 3);
-          const deltaY = Math.round((b.y - a.y) / 3);
-          // Update points
-          points[points.length - 1] = {
-            x: b.x,
-            y: b.y,
-            c: [
-              {
-                x: a.x + deltaX,
-                y: a.y + deltaY,
-              },
-              {
-                x: b.x - deltaX,
-                y: b.y - deltaY,
-              },
-            ],
-          };
-        }
-      }
-      const p = generatePath(points);
-      setPath(p);
-      setPathTxt(p);
-      setPoints([...points]);
-      setActivePoint(points.length - 1);
-    } else if (addText) {
-      const coords = getMouseCoords(e, false);
-      const txt = prompt("Enter text", "");
-      texts.push({ txt: txt, x: coords.x, y: coords.y });
-      setAddText(false);
-      setTexts(texts);
-    } else if (addAnchor) {
-      const coords = getMouseCoords(e, true);
-      anchors.push(coords);
-      setAddAnchor(false);
-      setAnchors(anchors);
-    } else if (addTradBelayStation) {
-      const coords = getMouseCoords(e, true);
-      tradBelayStations.push(coords);
-      setAddTradBelayStation(false);
-      setTradBelayStations(tradBelayStations);
-    }
-  }
-
-  function generatePath(points) {
-    let d = "";
-    points.forEach((p, i) => {
-      if (i === 0) {
-        // first point
-        d += "M ";
-      } else if (p.q) {
-        // quadratic
-        d += `Q ${p.q.x} ${p.q.y} `;
-      } else if (p.c) {
-        // cubic
-        d += `C ${p.c[0].x} ${p.c[0].y} ${p.c[1].x} ${p.c[1].y} `;
-      } else if (p.a) {
-        // arc
-        d += `A ${p.a.rx} ${p.a.ry} ${p.a.rot} ${p.a.laf} ${p.a.sf} `;
-      } else {
-        d += "L ";
-      }
-      d += `${p.x} ${p.y} `;
-    });
-    return d;
-  }
-
-  function handleMouseMove(e) {
-    e.preventDefault();
-    if (!shift && !addAnchor && !addText) {
-      if (draggedPoint) {
-        setPointCoords(getMouseCoords(e, true));
-      } else if (draggedCubic !== false) {
-        setCubicCoords(getMouseCoords(e, false), draggedCubic);
-      }
-    }
-    return false;
-  }
-
-  function handleMouseUp(e) {
-    // Remove selection caused by shift-button used to create new points
-    document.getSelection().removeAllRanges();
-    e.preventDefault();
-    // Overlap with an existing path if the two last points overlap
-    if (readOnlyPoints?.length >= 1 && points.length >= 2 && activePoint > 0) {
-      const p0 = points[activePoint - 1];
-      const p1 = points[activePoint];
-      const ix = readOnlyPoints?.findIndex((p) => p1.x === p.x && p1.y === p.y);
-      if (
```

Comment:
@jossi87 - I couldn't find out what this is intended to do (by playing around with the deployed version). Can you explain the intended use case to me? Then I can port it to the new implementation.

Reply:
[screen-capture.webm](https://github.com/jossi87/climbing-web/assets/17472908/08af4344-52af-4f8d-b75f-6b6f418cb203)
I think this is easier to explain with a video :)
It merges the path you are drawing with existing topo-paths on image (if the points you draw are close to an existing path). It also adds a point in the beginning if the first two points match the two first points on existing path. This is helpful while drawing topos like this one (where several lines share a start): https://brattelinjer.no/sector/2862?idMedia=20635
Is this helpful? Let me know if you have more questions. I'll leave this pull request open until I get feedback.

Reply:
That's very helpful - thanks! I'll get this ported now that I know what to look for.